frontend/app/controllers/sign-in.js

```diff
@@ -0,0 +1,68 @@
+import Ember from 'ember';
+
+export default Ember.Controller.extend({
+
+/*---- Login Screen ---*/
+
+  loginFailed: false,
+  loginFailedMessage: "Failed to Login",
+  isProcessing: false,
+  isSlowConnection: false,
+  timeout: null,
+
+    success: function(response) {
+         this.reset();
+         if (response.status==true){
+                this.reset();
+                window.location.href = "/userHome";
+         }
+         else{
+                this.set("loginFailed", true);
+                this.loginFailedMessage= String(response.message);
+         }
+    },
+    error: function() {
+         this.reset();
+         this.set("loginFailed", true);
+         this.set("loginFailedMessage",'Sign-In failed. Server down? ');
+    },
+    failure: function() {
+         this.reset();
+         this.set("loginFailed", true);
+         this.set("loginFailedMessage",'Sign-In failed. ? ');
+    },
+
+    slowConnection: function() {
+         this.set("isSlowConnection", true);
+    },
+
+    reset: function() {
+         clearTimeout(this.get("timeout"));
+         this.setProperties({
+              isProcessing: false,
+              isSlowConnection: false
+         });
+    },
+
+
+
+  actions:{
```

Comment:
@niranjan94 The [tutorial](http://blog.trackets.com/2013/05/23/how-to-write-a-login-form.html) did not include this nested dictionary called `actions` , but I got JS errors saying `logMeIn` action is not defined, so I moved the function inside a dict called `actions`, and it works fine. Is the method correct?
In the tutorial all the functions are simply defined inside `Ember.Controller.extend({ })` , and it seems to work fine for them.
I find this approach to be awkward. Maybe there's a better way?

I also found that I could also define this action `logMeIn` in app/routes/sign-in.js which saves the trouble of defining a controller, but defining persistent variables in routes/sign-in.js which I can access in the template does not seem to be possible. Please inform if it is.